algorithms/src/Kokkos_Random.hpp

```diff
@@ -1626,7 +1626,8 @@ template <class ExecutionSpace, class ViewType, class RandomPool,
 void fill_random(const ExecutionSpace& exec, ViewType a, RandomPool g,
                  typename ViewType::const_value_type begin,
                  typename ViewType::const_value_type end) {
-  Impl::apply_to_view_of_static_rank(
+  using ::Kokkos::Impl::apply_to_view_of_static_rank;
+  apply_to_view_of_static_rank(
```

Comment:
This avoid the need for a dependency on `kokkoscontainers` sicne there is a `apply_to_view_of_static_rank` overload for `DynRankView`.